src/App.tsx

```diff
@@ -14,11 +16,100 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string | null,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
+
+  decreaseGood = () => {
+    this.setState({
+      selectedGood: null,
+    });
+  };
+
+  selectGood = (good: string) => {
+    this.setState(prevState => {
+      if (prevState.selectedGood !== good) {
+        return {
+          selectedGood: good,
+        };
+      }
```

Comment:
you can handle only one case, don't need to do something if the good is the same